client-samples/dotnet/rest/api-template/services/ApiService.cs

```diff
@@ -0,0 +1,54 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+public class ApiService
+{
+    // Method to call an API with a given URL, token, and optional proxy
+    public static HttpResponseMessage GetResponse(string url, string token, string proxy)
+    {
+        Console.WriteLine($"Calling API at {url}.\n");
+        var httpClientHandler = new HttpClientHandler();
+
+        // Configure proxy if provided
+        if (!string.IsNullOrEmpty(proxy))
+        {
+            httpClientHandler.Proxy = new WebProxy(proxy);
+            httpClientHandler.UseProxy = true;
+        }
+
+        // Create an HttpClient instance with the configured handler
+        using var httpClient = new HttpClient(httpClientHandler);
+
+        // Set the Authorization header with the provided token
+        httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        
+        // Make a GET request
+        return httpClient.GetAsync(url).Result;

```

Comment:
I'd propose following through with keeping calls async.
So rather than calling .Result, the method could be "public static async Task<HttpResponseMessage> GetResponseAsync" and then await the call.
CallApi would then be "public static Task CallApi" and await GetResponse.